### PR TITLE
feat: auto-init

### PR DESCRIPTION
**Pull request by sxlijin**
Teach trunk-action to

* automatically create `.trunk/trunk.yaml` if none exists (i.e. `trunk init`) and
* install `npm`, `yarn`, or `pnpm` dependencies as appropriate

Add tests that provide sufficient coverage for a variety of scenarios.

Does not introduce caching - that will be separate.

### CODE REVIEW

File: .github/workflows/repo_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        with:
+          cache: false
+          check-mode: all
+          trunk-path: ${{ matrix.trunk-path }}
+          post-init: ${{ matrix.post-init }}
+        continue-on-error: true
+
+      - name: Check for task failures
+        shell: python
+        run: |
```
Comment (laurit17) [inline .github/workflows/repo_tests.yaml:119]: is it possible to extract this into its own file and have the env variables passed through as sys args?

Comment (sxlijin) [reply .github/workflows/repo_tests.yaml:119]: done

Comment (laurit17) [reply .github/workflows/repo_tests.yaml:120]: much better - thanks!

**Review by laurit17 — APPROVED** [review]
LGTM!

**Pull request merged**